pkg/operation/common/utils.go

```diff
@@ -428,6 +428,23 @@ func DeleteLoggingStack(ctx context.Context, k8sClient client.Client, namespace
 	return nil
 }
 
+// GetResourcesForContainerInStatefulSet  returns the current resources for container in the StatefulSet
+func GetResourcesForContainerInStatefulSet(ctx context.Context, k8sClient client.Client, key client.ObjectKey) ([]*corev1.ResourceRequirements, error) {
```

Comment:
As the function name suggests, I rather meant that you provide the container name and return only the resources for this container. But actually it's better like this because you only need one `Get()` call even if you are interested in the resources of multiple containers in the same statefulset. I just suggest to rename it one more time:
```suggestion
func GetContainerResourcesInStatefulSet(ctx context.Context, k8sClient client.Client, key client.ObjectKey) ([]*corev1.ResourceRequirements, error) {
```